File: src/services/TalentService.ts
import axios from "axios";
import Talent from "../models/Talent";
import {RootPath} from "./RootPath";

export default class TalentService {

    static async getTalents(): Promise<Talent[]> {
        return await (await axios.get(RootPath.Talent)).data;
    }

    static async getTalent(name: string): Promise<Talent> {
        return await (await axios.get(RootPath.Talent + name)).data;
    }

    static async createTalent(name: string): Promise<Talent> {
        return await (await axios.post(RootPath.Talent + name)).data;
    }

    static async updateTalent(name: string, talent: Talent): Promise<Talent> {
        return await (await axios.put(RootPath.Talent + name, talent)).data;
    }
}

File: gradle-java-server/src/main/java/com/github/genraven/gradlejavaserver/service/TalentService.java
package com.github.genraven.gradlejavaserver.service;

import com.github.genraven.gradlejavaserver.domain.Talent;
import com.github.genraven.gradlejavaserver.repository.TalentRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

@Service
public class TalentService {

    private final TalentRepository talentRepository;

    @Autowired
    public TalentService(final TalentRepository talentRepository) {
        this.talentRepository = talentRepository;
    }

    public Flux<Talent> getAllTalents() {
        return talentRepository.findAll();
    }

    public Mono<Talent> getTalent(final String name) {
        return talentRepository.findById(name);
    }

    public Mono<Talent> createTalent(final String name) {
        return talentRepository.save(new Talent(name));
    }

    public Mono<Talent> updateTalent(final String name, final Talent talent) {
        return getTalent(name).map(tal -> {
            tal.setActivation(talent.getActivation());
            tal.setRanked(talent.isRanked());
            tal.setTier(talent.getTier());
            tal.setDescription(talent.getDescription());
            tal.setSummary(talent.getSummary());
            tal.setModifiers(talent.getModifiers());
            tal.setSettings(talent.getSettings());
            return tal;
        }).flatMap(talentRepository::save);
    }
}
